compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1488,7 +1488,7 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         self.probe(|_| {
             // First check that the self type can be related.
             let sub_obligations = match self.at(&ObligationCause::dummy(), self.param_env).sup(
-                DefineOpaqueTypes::No,
+                DefineOpaqueTypes::Yes,
```

Comment:
we could try to get away with only using `DefineOPaqueTypes::No` here when checking whether the self type could be unified 🤔 I think all the other changes here are a lot less of a concern

Reply:
I manually added a check for that instead of relying on `No`. I'm surprised it doesn't work for the new solver. I don't see where it would replace a field expression's opaque type with an inference var.